src/us/kbase/workspace/test/kbase/HandleAndShockTest.java

```diff
@@ -391,15 +387,29 @@ private void checkMem(final Object num, final String name)
 			assertThat("bad " + name, (Long) num > 0, is(true));
 		}
 	}
-	
+
 	@SuppressWarnings("deprecation")
 	@Test
 	public void basicHandleTest() throws Exception {
 		String workspace = "basichandle";
 		CLIENT1.createWorkspace(new CreateWorkspaceParams().withWorkspace(workspace));
-		Handle h1 = HANDLE_CLIENT.newHandle();
+		BasicShockClient bsc = new BasicShockClient(
+				new URL("http://localhost:" + SHOCK.getServerPort()), CLIENT1.getToken());
+
+		final ShockNode node = bsc.addNode();
+		String shock_id = node.getId().getId();
+		final Handle handle = new Handle();
+		handle.setId(shock_id);
+		handle.setFileName("empty_file");
+		handle.setUrl(bsc.getShockUrl().toString());
+		handle.setType("shock");
+		String handle_id = UUID.randomUUID().toString();
+		System.out.println("generated random handle_id for testing: " + handle_id);
+		handle.setHid(handle_id);
```

Comment:
Does the handle service normally allow setting the ID? I thought the ID was an auto incrementing index key with the 'KBH_' prefix.

Reply:
👍 you are right. I don't need to set handle id.